src/pages/TopUp/components/WalletConnectModal.tsx

```diff
@@ -47,10 +67,14 @@ const WalletConnectModal: React.FC = () => {
     return !Object.values(AllowedNetworks).includes(network);
   }, [chainId]);
 
+  const handleClose = () => {
+    setLoading(false);
+  };
```

Comment:
I'm ok with it. But a nice to have:

Separate concerns a bit here:
- Lets control the `loading` state at the consumer level and pass only that state to the component.

```tsx
// at the consumer level

const handleWalletModalClose = () => {
  setLoading(false);
};

return <WalletConnectModal loading={loading} onClose={handleWalletModalClose} />

```

Reply:
Pushed a patch for this... removed the need to rename the above `_`